balancer/grpclb/grpclb.go

```diff
@@ -245,6 +259,34 @@ type lbBalancer struct {
 //  - does two layer roundrobin pick otherwise.
 // Caller must hold lb.mu.
 func (lb *lbBalancer) regeneratePicker() {
+	// Keep pickfirst logic in a separate if statement, even though it means
+	// some duplicate code.
+	if lb.usePickFirst {
+		switch lb.state {
+		case connectivity.Ready, connectivity.Idle:
+			readySCs := make([]balancer.SubConn, 0, 1)
```

Comment:
Hmm, this doesn't seem right to me - we shouldn't have more than one subconn when we are in pick first mode, right?  We should send the full list of addresses to that subconn and update it as necessary.  Am I misunderstanding the requirements, was that hard/impossible for some reason, or did you just not consider that approach?

Reply:
That's right, there's will only be one SubConn.
The following for loop breaks after the first iteration.

It's done this way to avoid rewriting the SubConn state change handling code.
And also we can reuse the pickers (both roundrobin and the fancy grpclb picker).